libpod/image.go

```diff
@@ -216,3 +217,45 @@ func (r *Runtime) GetImages(filter ...ImageFilter) ([]*storage.Image, error) {
 func (r *Runtime) ImportImage(path string) (*storage.Image, error) {
 	return nil, ctr.ErrNotImplemented
 }
+
+// FindImage searches for a *storage.Image with a matching the given name or ID in the given store.
+func (r *Runtime) FindImage(image string) (*storage.Image, error) {
+	var img *storage.Image
+	ref, err := is.Transport.ParseStoreReference(r.store, image)
+	if err == nil {
+		img, err = is.Transport.GetStoreImage(r.store, ref)
+	}
+	if err != nil {
+		img2, err2 := r.store.Image(image)
+		if err2 != nil {
+			if ref == nil {
+				return nil, errors.Wrapf(err, "error parsing reference to image %q", image)
+			}
+			return nil, errors.Wrapf(err, "unable to locate image %q", image)
+		}
+		img = img2
+	}
+	return img, nil
+}
+
+// FindImageRef searches for and returns a new types.Image matching the given name or ID in the given store.
+func (r *Runtime) FindImageRef(image string) (types.Image, error) {
```

Comment:
If we're calling the above GetImage, this should be GetImageRef

Reply:
Should also describe the difference between a types.Image and storage.Image somehow, otherwise this is a terrible API to work with. If possible we'd want to deal only with one or the other in the public API.

Reply:
@nalind WDYT?